File: src/main/java/net/igneo/icv/entity/boots/surfWave/SurfWaveModel.java
package net.igneo.icv.entity.boots.surfWave;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class SurfWaveModel extends GeoModel<SurfWaveEntity> {
    private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/surf.geo.json");
    private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/surf.png");
    private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/surf.animation.json");
    @Override
    public ResourceLocation getModelResource(SurfWaveEntity surfWaveEntity) {
        return model;
    }

    @Override
    public ResourceLocation getTextureResource(SurfWaveEntity surfWaveEntity) {
        return texture;
    }

    @Override
    public ResourceLocation getAnimationResource(SurfWaveEntity surfWaveEntity) {
        return animations;
    }
}

File: src/main/java/net/igneo/icv/entity/boots/surfWave/SurfWaveModel.kt
package net.igneo.icv.entity.boots.surfWave

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class SurfWaveModel : GeoModel<SurfWaveEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/surf.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/surf.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/surf.animation.json")

    override fun getModelResource(surfWaveEntity: SurfWaveEntity?): ResourceLocation {
        return model
    }

    override fun getTextureResource(surfWaveEntity: SurfWaveEntity?): ResourceLocation {
        return texture
    }

    override fun getAnimationResource(surfWaveEntity: SurfWaveEntity?): ResourceLocation {
        return animations
    }
}
